### PR TITLE
backups: expect a KeysBackupRequest

### PR DESCRIPTION
**Pull request by AndrewFerr**
Followup on https://github.com/matrix-org/matrix-rust-sdk/pull/2318

### CODE REVIEW

File: src/machine.rs [hunk 1]
```diff
@@ -918,16 +918,16 @@ impl OlmMachine {
...
 
         future_to_promise(async move {
-            let request = me.backup_machine().backup().await?;
-
-            let Some(request) = request else {
-                return Ok(None);
-            };
+            match me.backup_machine().backup().await? {
+                Some((transaction_id, keys_backup_request)) => {
+                    Ok(Some(JsValue::from(requests::KeysBackupRequest::try_from((
```
Comment (richvdh) [inline src/machine.rs:923]: not sure we need the explicit `JsValue::from` here? I think `future_to_promise` will work with anything which implements `Into<JsValue>` (as `KeybackupRequest does`) ?

File: src/machine.rs [hunk 2]
```diff
@@ -918,16 +918,16 @@ impl OlmMachine {
         let me = self.inner.clone();
 
         future_to_promise(async move {
-            let request = me.backup_machine().backup().await?;
-
-            let Some(request) = request else {
-                return Ok(None);
-            };
+            match me.backup_machine().backup().await? {
```
Comment (richvdh) [inline src/machine.rs:921]: I preferred the `let else` over the `match` :stuck_out_tongue:.

YMMV, happy for you to change it.

File: src/machine.rs (hunk 1 above)
Comment (AndrewFerr) [reply src/machine.rs:923]: Ah that's nice, I'll get rid of it then.

File: src/machine.rs (hunk 2 above)
Comment (AndrewFerr) [reply src/machine.rs:921]: I went with the `match` to format this similarly to `OlmMachine.get_missing_sessions`, which is what gave me the idea of working with a `(request_id, request)` tuple in the first place.  Thanks for letting me keep it :slightly_smiling_face:

**Pull request merged**